src/SonarScanner.MSBuild.PreProcessor/PreProcessor.cs

```diff
@@ -157,7 +157,8 @@ private async Task<ArgumentsAndRuleSets> FetchArgumentsAndRuleSets(ISonarWebServ
 
                     // Generate Roslyn analyzers settings and rulesets
                     // It is null if the processing of server settings and active rules resulted in an empty ruleset
-                    var analyzerProvider = factory.CreateRoslynAnalyzerProvider(server);
+                    var localCacheTempPath = args.GetSetting(SonarProperties.PluginCacheDirectory, Path.Combine(Path.GetTempPath(), ".sonarqube", "resources"));
```

Comment:
`Path.Combine(Path.GetTempPath(), ".sonarqube", "resources")` duplicates the logic from `EmbeddedAnalyzerInstaller.GetLocalCacheDirectory`

Reply:
Yes it does indeed, but I did that (slightly reluctantly) as I didn't want to add an if statement in `CreateRoslynAnalyzerProvider` of `PreprocessorObjectFactory`.

What I did end up doing was change the constructor for the `EmbeddedAnalyzerInstaller` to not throw when no path was specified.